internal/dot/graph.go

```diff
@@ -25,15 +25,26 @@ import (
 	"reflect"
 )
 
+// State of a constructor or group is updated when they fail to build.
+type State int
+
+const (
+	initial State = iota
+	pointOfFailure
+	transitiveFailure
+)
+
 // Ctor encodes a constructor provided to the container for the DOT graph.
 type Ctor struct {
 	Name        string
 	Package     string
 	File        string
 	Line        int
+	ID          uintptr
```

Comment:
Instead of using raw `uintptr`s, can we declare a type?

    type CtorID uintptr

From the main dig package, you'll do `dot.CtorID(cptr)` to build this.